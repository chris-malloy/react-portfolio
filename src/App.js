import React, { Component } from 'react';
import './App.css';
import MyNavbar from './components/MyNavbar';
import Header from './components/Header';
import Projects from './components/Projects';

class App extends Component {
  render() {
    return (
      <div className="blue">
        <MyNavbar />
        <div className="container">
          <Header />
          <h5 className="center">Projects</h5>
          <Projects />
        </div>
      </div>
    );
  }
}

export default App;
