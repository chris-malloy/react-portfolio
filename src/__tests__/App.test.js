import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App';

configure({ adapter: new Adapter() });

describe('App', () => {
  const app = shallow(<App />);

  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });
  describe('my components', () => {
    it('contains a `Title` component', () => {
      expect(app.find('Title').exists()).toBe(true);
    });

    it('contains a `MyNavbar` component', () => {
      expect(app.find('MyNavbar').exists()).toBe(true);
    });

    it('contains a `Header` component', () => {
      expect(app.find('Header').exists()).toBe(true);
    });

    it('contains a `Projects` component', () => {
      expect(app.find('Projects').exists()).toBe(true);
    });

    it('contains a `Skills` component', () => {
      expect(app.find('Skills').exists()).toBe(true);
    });

    it('contains a `MyFooter` component', () => {
      expect(app.find('MyFooter').exists()).toBe(true);
    })
  });
});
