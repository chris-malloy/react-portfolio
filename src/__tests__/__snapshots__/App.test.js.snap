// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MyNavbar />,
        <div
          className="container"
>
          <Header />
          <Title
                    id="projects"
                    title="Projects"
          />
          <Projects />
          <Title
                    id="skills"
                    title="Skills"
          />
          <Skills />
          <MyFooter />
</div>,
      ],
      "className": "blue",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header />,
            <Title
              id="projects"
              title="Projects"
/>,
            <Projects />,
            <Title
              id="skills"
              title="Skills"
/>,
            <Skills />,
            <MyFooter />,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "projects",
              "title": "Projects",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "skills",
              "title": "Skills",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MyNavbar />,
          <div
            className="container"
>
            <Header />
            <Title
                        id="projects"
                        title="Projects"
            />
            <Projects />
            <Title
                        id="skills"
                        title="Skills"
            />
            <Skills />
            <MyFooter />
</div>,
        ],
        "className": "blue",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Header />,
              <Title
                id="projects"
                title="Projects"
/>,
              <Projects />,
              <Title
                id="skills"
                title="Skills"
/>,
              <Skills />,
              <MyFooter />,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "projects",
                "title": "Projects",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "skills",
                "title": "Skills",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
